Add `set_sample_data` keys and explain how they work (#314)
appsignal/appsignal-docs: source/nodejs/instrumentation/span.html.md

  .addError(new Error("test error"))
```

## `span.set(key: string, value: string | number | boolean)`

Sets arbitary data on a current `Span`. The data can be a `string`, `number` or `boolean` type.

## `span.setNamespace(name: string)`

Sets the `namespace` of the current `Span`. Namespaces are a way to group error incidents, performance incidents, and host metrics in your app.

## `span.addError(error: Error | object)`

Sets the `error` of the current `Span`. When an `Error` object is passed to the `setError` method, the `stack` property is normalised and transformed into an array of strings, with each string representing a line in the stacktrace.

> ## `span.setSampleData(key: string, data: Array<any> | { [key: string]: any })`

Reviewer: ```suggestion
## `span.setSampleData(key: string, data: Array<string | number | boolean> | { [key: string]: string | number | boolean })`
```

Fixing my own work here!

Adds sample data to the current `Span`. The sample data object must not contain any nested objects. This call accepts the following keys, some keys require a specific format.

> ### `sesssion_data`

Reviewer: ```suggestion
### `session_data`
```

Filled with session/cookie data by default, but can be overridden with the following call:

```
span.setSampleData("sesssion_data", {_csrf_token: "Z11CWRVG+I2egpmiZzuIx/qbFb/60FZssui5eGA8a3g="})
```

This key accepts nested objects that will be rendered as JSON on a Incident Sample page for both Exception and Performance samples.

![sesssion_data](/assets/images/screenshots/sample_data/session_data.png)

### `params`

> Filled with framework (such as Phoenix) parameters by default, but can be overridden or filled with the following call:

Reviewer: ```suggestion
Filled with framework (such as Express) parameters by default, but can be overridden or filled with the following call:
```

```

> span.setSampleData("params", {action: "show", controller: "homepage"})

Reviewer: ```suggestion
span.setSampleData("params", {string: "value", number: 123 })
```

```

This key accepts nested objects and will show up as follows on a Incident Sample page for both Exception and Performance samples, formatted as JSON.

![params](/assets/images/screenshots/sample_data/params.png)

### `environment`

> Environment variables from a request/background job, filled by the Phoenix integration, but can be filled/overriden with the following call:

Reviewer: ```suggestion
Environment variables from a request/background job (typically filled by the default `http` integration, but can be further augmented by other integrations), but can be filled/overriden with the following call:
```
